lookout/core/lib.py

```diff
@@ -86,21 +91,57 @@ def files_by_language(files: Iterable[File]) -> Dict[str, Dict[str, File]]:
     return result
 
 
-def filter_files(files: Dict[str, File], line_length_limit: int,
-                 log: logging.Logger = None) -> List[File]:
+def filter_filepaths(filepaths: Iterable[str], exclude_pattern: Optional[str] = None,
+                     ) -> Iterable[str]:
     """
-    Filter files based on their maximum line length.
+    Mirror of the file filtering used in the format analyzer for use by debugging tools.
 
-    :param files: files to filter.
+    :param filepaths: Iterable of filepaths to filter.
+    :param exclude_pattern: Pattern to reject files based on their path. If None, uses the pattern
+                            currently in use in lookout.core. Use "" to not filter anything.
+    :return Iterable of paths, filtered.
+    """
+    if exclude_pattern is None:
+        exclude_pattern = GARBAGE_PATTERN
+    exclude_compiled_pattern = re.compile(exclude_pattern) if exclude_pattern else None
+    for filepath in filepaths:
+        if not isfile(filepath):
+            continue
+        if exclude_compiled_pattern and exclude_compiled_pattern.search(filepath):
+            continue
+        yield filepath
+
+
+def filter_files(filenames: Iterable[str], line_length_limit: int, client: BblfshClient,
+                 language: str, log: logging.Logger = None) -> Iterable[File]:
+    """
+    Filter files based on `language` and their maximum line length.
+
+    :param filenames: paths to the files to filter.
     :param line_length_limit: maximum line length to accept a file.
+    :param client: Babelfish client. Babelfish server should be started accordingly.
+    :param language: Language to consider. Will discard the other languages.
     :param log: logger to use to report the number of excluded files.
     :return: files passed through the filter and the number of files which were excluded.
     """
     passed = []
-    for file in files.values():
-        if len(max(file.content.splitlines(), key=len, default=b"")) <= line_length_limit:
-            passed.append(file)
+    n_parsed = 0
+    for filename in tqdm(filter_filepaths(filenames)):
```

Comment:
We cannot do this progress tracking. This is a library, it does not care about the details.

Add another argument which would be `lambda x: x` by default and replace `tqdm`.